### DISCUSSION TITLE
If I made an API call inside getInitialProps will this API get called every time the user enters the page?

### DISCUSSION THREAD

**Discussion by bymoe** [vercel/next.js#14208] [Help]
As simple as the example in the docs...
![error](https://user-images.githubusercontent.com/20565199/84714380-27fb3e00-af76-11ea-8854-cec832375cf3.png)
Does this API get called every time the client enters the page on client-side?

**If no, Is there something similar in Next.js so that in my use case I want to wait for client-side API call before rendering the page?**

**Comment by coetry** [comment]
Waiting for the client to resolve the API request before rendering would be done with React (Next.js doesn't expose anything special for this). You can do something similar to:

```js
function Page() {
  const [data, setData] = useState()
 
  useEffect(() => {
     async function fetchData() {
       // data fetching logic
       // once data is retrieved, call setData() w/ data to update state
     }
    fetchData()
  })
  if (!data) return <div>Loading ...</div>
  return <div>Cool page with {data}</div>
}

**Comment by srikanth912** [comment]
Yes, it does call API every time irrespective of whether rendering is happening on the server-side (first request) or client-side (subsequent requests).
